### PR TITLE
feat(profiles): Support profiler_id in context

### PR DESCRIPTION
**Pull request by phacops**
As part of continuous profiling, we need to store a `profiler_id` on the profile context of a transaction.

### CODE REVIEW

**Review by Dav1dde — COMMENTED** [review]
This won't work, the context is recreated in `transfer_id`:
https://github.com/getsentry/relay/blob/bdb89f2318a3f28d9212d4faf03b0634c329b8c9/relay-server/src/services/processor/profile.rs#L66-L87

File: relay-event-schema/src/protocol/contexts/profile.rs [hunk 1]
```diff
@@ -10,8 +10,12 @@ use crate::protocol::EventId;
 #[cfg_attr(feature = "jsonschema", derive(JsonSchema))]
 pub struct ProfileContext {
     /// The profile ID.
-    #[metastructure(required = "true")]
+    #[metastructure(required = "false")]
     pub profile_id: Annotated<EventId>,
+
+    /// The profiler ID.
+    #[metastructure(required = "false")]
+    pub profiler_id: Annotated<EventId>,
```
Comment (jjbayer) [inline relay-event-schema/src/protocol/contexts/profile.rs:16]: Would we transfer this `profiler_id` from the profile itself, just like we do for `profile_id`?

Comment (phacops) [reply relay-event-schema/src/protocol/contexts/profile.rs:16]: `profiler_id` wouldn't be transferred. Either it's there or it's not.

File: relay-event-schema/src/protocol/contexts/profile.rs [hunk 2]
```diff
@@ -10,8 +10,12 @@ use crate::protocol::EventId;
 #[cfg_attr(feature = "jsonschema", derive(JsonSchema))]
 pub struct ProfileContext {
     /// The profile ID.
-    #[metastructure(required = "true")]
+    #[metastructure(required = "false")]
```
Comment (Dav1dde) [inline relay-event-schema/src/protocol/contexts/profile.rs]: Isn't required = false the default and we should just omit the entire thing?

File: relay-server/src/services/processor/profile.rs [hunk 3]
```diff
@@ -76,13 +76,10 @@ pub fn transfer_id(
...
             contexts.add(ProfileContext {
                 profile_id: Annotated::new(profile_id),
+                ..Default::default()
             });
         }
-        None => {
-            if let Some(contexts) = event.contexts.value_mut() {
-                contexts.remove::<ProfileContext>();
-            }
-        }
```
Comment (Dav1dde) [inline relay-server/src/services/processor/profile.rs]: Now it's possible that we emit a profile id without a corresponding profile, the code was introduced to exactly prevent that.

File: relay-server/src/services/processor/profile.rs [hunk 4]
```diff
@@ -76,13 +76,10 @@ pub fn transfer_id(
...
+        _ => (),
```
Comment (Dav1dde) [inline relay-server/src/services/processor/profile.rs]: If that is the intended behaviour `if let Some() = profile_id` is the thing to use here.

**Review by Dav1dde — APPROVED** [review]

File: relay-server/src/services/processor/profile.rs [hunk 5]
```diff
@@ -76,11 +76,14 @@ pub fn transfer_id(
...
                 profile_id: Annotated::new(profile_id),
+                ..ProfileContext::default()
             });
         }
         None => {
             if let Some(contexts) = event.contexts.value_mut() {
-                contexts.remove::<ProfileContext>();
+                if let Some(profile_context) = contexts.get_mut::<ProfileContext>() {
+                    profile_context.profile_id = Annotated::empty();
+                }
```
Comment (Zylphrex) [inline relay-server/src/services/processor/profile.rs:86]: A test for this would be nice that only the `profile_id` (not `profiler_id`) gets removed.

**Review by Zylphrex — APPROVED** [review]
LGTM

File: relay-event-normalization/src/normalize/span/tag_extraction.rs [hunk 6]
```diff
@@ -132,6 +134,8 @@ impl SpanTagKey {
             SpanTagKey::TraceStatus => "trace.status",
             SpanTagKey::MessagingDestinationName => "messaging.destination.name",
             SpanTagKey::MessagingMessageId => "messaging.message.id",
+            SpanTagKey::ThreadName => "thread.name",
+            SpanTagKey::ThreadId => "thread.name",
```
Comment (Dav1dde) [inline relay-event-normalization/src/normalize/span/tag_extraction.rs]:
```suggestion
            SpanTagKey::ThreadId => "thread.id",
```

**Review by Dav1dde — COMMENTED** [review]
Please move the thread id/name changes to a separate PR

**Pull request merged**